Reconstructed Code & Object classes, added Scene class

frankhjwx opened the pull request:
removed 'easing' variable in objects, and we can add commands more humanlikely.
added scene class for group command changes. 
e.g.:
M(2,1000,2000,320,240,330,250) √  ->  M,2,1000,2000,320,240,330,250
M(1000,2000,320,240,330,250) √  -> M,0,1000,2000,320,240,330,250
M(1000,2000,320,240) √ ->  M,0,1000,2000,320,240
M(1000,320,240) √ -> M,0,1000,,320,240

added remove func from 0.0.2 branch to support remove commands.
e.g. obj.remove('M') will remove all M commands from obj.

frankhjwx committed:
add support for removing commands

frankhjwx committed:
Start a new branch for Logic work

frankhjwx committed:
Revert "Start a new branch for Logic work"

This reverts commit 907413b5b2cce8e7e4b3c3c300ff0576a9247d85.

frankhjwx committed:
Update support for easy use of easing, and added class StoryboardScene

in class scene, we can do batch operations on a group of objects

frankhjwx committed:
add Remove func

Mahua233 commented on Storyboard Engine/StoryboardObject.py:
```diff
@@ -6,12 +6,12 @@
 # Old Class, will be modified later
 class Object:
-    def __init__(self, file_name, type='Sprite', origin='Centre', x=320, y=240,
+    def __init__(self, file_name, layer='Foreground', type='Sprite', origin='Centre', x=320, y=240,
```
exchange layer and type? 'Sprite,"layer","origin","filepath",x,y' <- look at is

Mahua233 commented on Storyboard Engine/StoryboardObject.py:
```diff
@@ -42,107 +42,184 @@ def add(self, x):
+        else:
+            raise RuntimeError('Not in any loop!')
+
+    def Remove(self, key):
```
Add a Remove by index func?

Mahua233 reviewed (commented):
http://zh-google-styleguide.readthedocs.io/en/latest/google-python-styleguide/python_style_rules/#id16 好像考试大师你说要用规范来着 这边写着推荐Method Names lower_with_under()(Public) _lower_with_under() (protected) __lower_with_under() (private)

frankhjwx commented:
等oriental blossom搞完就修

frankhjwx committed:
Merge branch 'Develop-Yumeno_Himiko-0.01' of https://github.com/frankhjwx/OSU-Storyboard-Engine into Develop-Yumeno_Himiko-0.01

frankhjwx committed:
support sb elements straight write into files

Mahua233 reviewed (approved):
ok

frankhjwx committed:
Rewrite some classes in LyricsParser

now split the usage of CharacterRenderer and LyricParser, and support add single sentences using commands

Mahua233 committed:
Change type of function.

Mahua233 committed:
Fixed something not important.
